src/Core/Generator/SchemaGeneratorFactory.cs

```diff
@@ -56,46 +58,89 @@ public static async Task<string> Create(RuntimeConfig config, string mode, int?
+            HashSet<string> dbAndContainerToProcess = new();
+            if (globalDatabaseName != null && globalContainerName != null)
+            {
+                dbAndContainerToProcess.Add($"{globalDatabaseName}.{globalContainerName}");
+            }
 
-            // Determine the appropriate sampler based on the sampling mode.
-            ISchemaGeneratorSampler schemaGeneratorSampler = samplingMode switch
+            foreach (KeyValuePair<string, Entity> entity in config.Entities)
             {
-                SamplingModes.TopNExtractor => new TopNExtractor(container, sampleCount, days, logger),
-                SamplingModes.EligibleDataSampler => new EligibleDataSampler(container, partitionKeyPath, sampleCount, days, logger),
-                SamplingModes.TimePartitionedSampler => new TimePartitionedSampler(container, groupCount, sampleCount, days, logger),
-                _ => throw new ArgumentException($"Invalid sampling mode: {mode}, Valid Sampling Modes are: {SamplingModes.TopNExtractor}, {SamplingModes.EligibleDataSampler}, {SamplingModes.TimePartitionedSampler}")
-            };
+                string entitySourceObject = entity.Value.Source.Object;
 
-            logger.LogInformation("Sampling Started using {0}", schemaGeneratorSampler.GetType().Name);
+                string[]? databaseContainerInfo = entitySourceObject?.Split('.');
+                string result = databaseContainerInfo switch
+                {
+                    null or { Length: 0 } when globalDatabaseName != null && globalContainerName != null => $"{globalDatabaseName}.{globalContainerName}",
+                    { Length: 2 } => entity.Value.Source.Object,
+                    { Length: 1 } when globalDatabaseName != null => $"{globalDatabaseName}.{entity.Value.Source.Object}",
+                    _ => throw new InvalidOperationException("Unexpected Source format")
+                };
 
-            // Get sample data from the selected sampler.
-            List<JsonDocument> dataArray = await schemaGeneratorSampler.GetSampleAsync();
+                dbAndContainerToProcess.Add(result);
+            }
 
-            logger.LogInformation("{0} records collected as Sample", dataArray.Count);
-            if (dataArray.Count == 0)
+            StringBuilder schema = new();
+            foreach (string dbAndContainer in dbAndContainerToProcess)
             {
-                logger.LogError("No data was sampled. Please try a different sampling mode or provide different sampling options.");
-                throw new ArgumentException("No data was sampled. Please try a different sampling mode or provide different sampling options.");
-            }
+                if (string.IsNullOrEmpty(dbAndContainer)) // Safety check
+                {
+                    continue;
+                }
+
+                string[] dbContainer = dbAndContainer.Split('.');
+                if (dbContainer.Length != 2)
```

Comment:
With the fix in the latest iteration using the switch statement, Length will always be == 2, right? otherwise you would have thrown an exception already on line 90.. Is this check needed here?